### ISSUE TITLE
Electron crashed

### ISSUE THREAD

**Issue by frozeman** [atom/electron#4021]
Electron just crashed for me while running my app and i'm not sure what could cause it.
Here is the log:

https://gist.github.com/frozeman/59da4a8213ff5a6c1bf7

**Comment by alexvandesande** [comment]
Same here. Using mac, didn't touch the app for a few minutes and then suddenly it crashed..

[log](https://gist.github.com/alexvandesande/a1c023dab2b059e797eb)

**Comment by zcbenz** [comment]
What's the version of Electron?

**Comment by frozeman** [comment]
0.36.2

**Comment by frozeman** [comment]
@zcbenz if you can point me to the part which crashed, i can probably figure out what in my code is causing it.

Is it something with the IPC and closed web contents? Or that i try to access `web contents.getId()`, after its gone?
Any hint?

**Comment by frozeman** [comment]
BTW, this didn't happen before 0.35.0

**Comment by zcbenz** [comment]
Did you create BrowserWindow in renderer process in your code like this?

```javascript
var BrowserWindow = remote.BrowserWindow;
var w = new BrowserWindow;
```

**Comment by frozeman** [comment]
No, i don't create BrowserWindows from inside the render process, but i have multiple webviews, which i think are implemented similarly

**Comment by zcbenz** [comment]
@frozeman Does your app's code call `getId` anywhere? I investigated all the places in Electron that call `getId`, none of them match the stack trace.

From the stack trace your code is calling `webContents.getId` after the webContents gets destroyed, the crash line should be in the read listener of a stream. Anyway I'm going to fix the crash, so when the same thing happens a JS exception would be thrown instead of crashing the whole program.

**Comment by frozeman** [comment]
Yes i do call `webContents.getId`, and it threw an exception before, which i catches with a try catch.
In this new version, i crashed.

Thanks for fixing :)

**Comment by frozeman** [comment]
@zcbenz is this fix in the latest 0.36.4 release?

**Comment by zcbenz** [comment]
Yeah 0.36.4 fixed this.
